### PR TITLE
Support Https

### PR DESCRIPTION
**Pull request by samgiles**

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ impl Db {
 
     pub fn seconds_from_epoch() -> i64 {
         let now = SystemTime::now();
-        now.duration_since(UNIX_EPOCH).unwrap().as_secs() as i64
+        now.duration_from_earlier(UNIX_EPOCH).unwrap().as_secs() as i64
```
Comment (samgiles) [inline src/db.rs]: `duration_since` has been renamed to `duration_from_eariler` in nightly.

Comment (samgiles) [inline src/db.rs]: Maybe not.. I was actually using an older version.

File: src/main.rs [hunk 2]
```diff
@@ -73,9 +78,30 @@ fn main() {
...
+    } else {
+        info!("Starting TLS server");
+        let certificate_directory = args.flag_cert_directory.unwrap();
+        let certificate_directory = PathBuf::from(certificate_directory);
+
+        let mut private_key = certificate_directory.clone();
+        private_key.push("privkey.pem");
+
+        let mut cert = certificate_directory.clone();
+        cert.push("cert.pem");
```
Comment (michielbdejong) [inline src/main.rs:99]: what about the chain cert? I'm seeing an error when trying `openssl s_client -connect knilxof.org:4443` (which runs this code) but not when trying `openssl s_client -connect knilxof.org:4243` (which runs node proxy.js). Also, that reminds me, we can remove proxy.js from this repo once this is resolved.

**Pull request merged**